owid/catalog/catalogs.py

```diff
@@ -63,13 +67,15 @@ class LocalCatalog(CatalogMixin):
     """
 
     path: Path
+    frame: "CatalogFrame"
 
     def __init__(self, path: Union[str, Path]) -> None:
         self.path = Path(path)
-        if self._catalog_file.exists():
-            self.frame = CatalogFrame(pd.read_feather(self._catalog_file.as_posix()))
-        else:
-            self.frame = CatalogFrame.create_empty()
+        if not self._catalog_file.exists():
+            self.reindex()
+
+        self.frame = CatalogFrame(pd.read_feather(self._catalog_file.as_posix()))
```

Comment:
Maybe I am missing something, but I would put an `else` clause here. Otherwise, if `not self._catalog_file.exists()` evaluate to `True`, we read the catalog from disk even though we already have it loaded in memory (as attribute `frame`).


```python
if not self._catalog_file.exists():
    self.reindex()
else:
    self.frame = CatalogFrame(pd.read_feather(self._catalog_file.as_posix()))
```

Reply:
Good catch, I added the `else` clause and fixed a missing `_base_uri` in the `reindex()` method too.

Reply:
LGTM 😄